src/main/oAuthClient.ts

```diff
@@ -276,6 +286,54 @@ export default class OAuthClient {
     })
   }
 
+  private loginWithIdToken(provider: string, idToken: string, nonce: string, opts: AuthOptions = {}): Promise<void> {
+    const authParams = this.authParams({
+      ...opts,
+      useWebMessage: false
+    })
+
+    return this.loginWithRedirect({
+      ...authParams,
+      provider,
+      idToken,
+      nonce,
+    })
+  }
+
+  private googleOneTap(opts: AuthOptions = {}): void {
+    if (this.config?.googleClientId) {
+      const nonce = randomBase64String()
```

Comment:
We should enable SDK users to pass their own `nonce` and only generate one if none is provided.